docs/API_SPECS_DEVICE_MODEL_SCRIPTS.md

```diff
@@ -0,0 +1,183 @@
+    "$created": "2018-07-31T21:24:24+00:00",
+    "$modified": "2018-07-31T21:24:24+00:00"
+  }
+}
+```
+
+### Editing with PUT
+
+When invoking the API using the PUT HTTP method, the service will attempt
+to modify an existing device model script. When using PUT, the device model 
+script Id is passed through the URL. PUT requests are idempotent and don't
+generate errors when retried (unless the payload differs during a retry, 
+in which case the ETag mismatch will generate an error).
+
+```
+PUT /v1/devicemodelscripts/53009673-6c49-4514-9dbd-5f811723c195 HTTP/1.1
+Host: localhost:9003
+Content-Type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW
+Cache-Control: no-cache
+Postman-Token: f5b9c0b9-1ce7-4996-a5e1-431967fc3aa4
+
+------WebKitFormBoundary7MA4YWxkTrZu0gW
+Content-Disposition: form-data; name="file"; filename="Reboot-method.js"
+Content-Type: application/javascript
+
+
+------WebKitFormBoundary7MA4YWxkTrZu0gW
+Content-Disposition: form-data; name="Etag"
+
+"b20375e9-0000-0000-0000-5b5914650000"
+------WebKitFormBoundary7MA4YWxkTrZu0gW--
+```
+```json
+{
+  "ETag": "\"01004b86-0000-0000-0000-0105b6004c86\"",
+  "Id": "b62d3316-effe-41d4-8767-e0ca6d07f013",
+  "Type": "javascript",
+  "Name": "Reboot-method.js",
+  "Content": "// Copyright (c) Microsoft. All rights reserved.\n\n/*global log*/\n/*global updateState*/\n/*global sleep*/\n/*jslint node: true*/\n\n\"use strict\";\n\n// Default state\nvar state = {\n    // reboot just changes whether the device is on or offline\n    online: true\n};\n\n/**\n * Entry point function called by the method.\n *\n * @param context        The context contains current time, device model and id\n * @param previousState  The device state since the last iteration\n * @param previousProperties  The device properties since the last iteration\n */\n/*jslint unparam: true*/\nfunction main(context, previousState, previousProperties) {\n\n    // Reboot - devices goes offline and comes online after 20 seconds\n    log(\"Executing 'Reboot' JavaScript method simulation.\");\n\n    state.DeviceMethodStatus = \"Rebooting device...\";\n    state.CalculateRandomizedTelemetry = false;\n    state.online = false;\n    // update the state to offline\n    updateState(state);\n\n    // Sleep for 15 seconds\n    sleep(15000);\n\n    state.DeviceMethodStatus = \"Successfully rebooted device.\";\n    updateState(state);\n\n    // Sleep for 5 seconds\n    sleep(5000);\n    state.CalculateRandomizedTelemetry = true;\n    // update the state back to online\n    state.online = true;\n    state.DeviceMethodStatus = \"\";\n    updateState(state);\n\n    log(\"'Reboot' JavaScript method simulation completed.\");\n}\n",
+  "Path": "Storage",
+  "$metadata": {
+    "$type": "SimulationScript;1",
+    "$uri": "/v1/simulationscripts/b62d3316-effe-41d4-8767-e0ca6d07f013",
+    "$created": "2018-07-31T21:24:24+00:00",
+    "$modified": "2018-07-31T22:24:24+00:00"
+  }
```

Comment:
for simplicity please remove the metadata from the request, these fields are not required (and ignored)

Reply:
Done.